### DISCUSSION TITLE
How to use addFiles function?

### DISCUSSION THREAD

**Discussion by yuvrajchaudhari03** [excalidraw/excalidraw#4351] [Q&A]
I got my answer thanks

**Answer**

Never call addFiles function without passing any parameter or else it will throw an error and crash the app

`excalidrawRef.current.addFile(ImgData)`

where **ImgData** is

`[
    {
        created: 1638377062740,
        dataURL: "data:image/jpeg;base64,/9j/4AAQSkZJR....",
        id: "5dd622b1f75815194a2d87a8f104d60519776600",
        mimeType: "image/jpeg",
    },
    {
        created: 1638377337782,
        dataURL: "data:image/jpeg;base64,/9j/4AAQSk......",
        id: "f77f811429dc41ac4460b065ed4f6e6cc7a5e213",
        mimeType: "image/jpeg",
    }
]`

**Comment by zsviczian** [comment]
This is how I use AddFiles in the Obsidian-Excalidraw plugin

https://github.com/zsviczian/obsidian-excalidraw-plugin/blob/1a2e7ac23fe1bddb11b8fc8f4b675d69dd531742/src/ExcalidrawView.ts#L87

https://github.com/zsviczian/obsidian-excalidraw-plugin/blob/1a2e7ac23fe1bddb11b8fc8f4b675d69dd531742/src/EmbeddedFileLoader.ts#L206

**Comment by dwelle** [comment]
> Never call addFiles function without passing any parameter or else it will throw an error and crash the app

@yuvrajchaudhari03 I'd recommend using TypeScript. We generally don't write defensive code that works with invalid input.

**Comment by programeGreenhand** [comment]
so how can I use this function to add my image to convas ????Why I use addFiles that is error and image become a gray image that is uncommon image
